### ISSUE TITLE
Brightness effect

### ISSUE THREAD

**Issue by jpcima** [rerrahkr/BambooTracker#118]
~~When volume of a FM channel is modified, it will affect the 4 operators regardless of algorithm.
Maybe it is more appropriate to modify the TL on *carrier* operators only, because modulators affect the timbre.~~

~~When I checked, this is the way how Deflemask operated; and it's also the way of implementing in libOPNMIDI.~~

The fact of affecting TL of *modulators* only, as opposed to the carriers, can serve to implement a useful effect ("brightness") similar to a lowpass cutoff control.
(example https://www.youtube.com/watch?v=-PRagjJsSq0)

EDIT: Could this kind of effect be provided?

**Comment by rerrahkr** [comment]
That is strange.
https://github.com/rerrahkr/BambooTracker/blob/84608f996cc6c21cb2539e3f6eaa0e2acaa5ddd0/BambooTracker/opna_controller.cpp#L938-L943
I should have checked if an operator is a career by `OPNAController::isCareer` in `OPNAController::writeFMEnvelopeToRegistersFromInstrument` and `OPNAController::writeFMEnveropeParameterToRegister` before write the value to the register...

**Comment by jpcima** [comment]
It's a misexamination of the register dump from me at yesterday night.. my apologies for noise; volume register were set, but values identical. My bad :man_facepalming:

I'll change this to feature request: I've wished to use the "brightness" effect, is this a possibility for BambooTracker? this is a TL change applied to the modulators, instead of the carriers.

**Comment by rerrahkr** [comment]
Since I think to add some "special effects" like Deflemask's Exx in the future, I will try to add FM brightness effects to them.

**Comment by rerrahkr** [comment]
I will add the TL relative set effect instead of the brightness effect.
I think that it is not clear which career changes with the brightness effect. For example, [the algorithm is set to 2](https://github.com/rerrahkr/BambooTracker/wiki/Sound-Chip#parameters), There are 7 combinations of careers to modulate (1, 2, 3, 1-2, 1-3, 2-3, 1-2-3).

`Bxyy` the TL relative set effect will have 2 parameters:

1. `x` - Target operator flag. Bit 0 is operator 1 and bit 3 is operator 4.
2. `yy` - Relative TL value (yy - $80).

**Comment by rerrahkr** [comment]
> I will add the TL relative set effect instead of the brightness effect.

I've tested it, but it's inconvenient to specify an operator each time, so I will implement the brightness effect that changes only the modulator directly connected to the carriers.

**Comment by rerrahkr** [comment]
I added brightness effect `0Bxx` at 60f28d9474c20d5800b9f951edd1ba08df3c4e66.

**Comment by jpcima** [comment]
Thank you, I will give a try sometime later.
Just now, I noticed a typo in `effect_description.cpp` : "relavtive" (relative)

**Comment by rerrahkr** [comment]
Ah, it's my trivial mistake, thank you for your report.

**Comment by jpcima** [comment]
`getOperatorsInLevel` is mistaken.

In the level 0, algorithm 7: It must be Operator number 3, and not 4.
```
                case 7:
                        return { 0, 1, 2, 4 };
```

In the level 3, algorithm 0: It must be Operator number 0, and not 1.

```
                case 0:
                        return { 1 };
```

**Comment by rerrahkr** [comment]
Sorry, I misunderstood the operator number. fixed at 497621cdaf0562f2e4536c731acdcf8f7bb35c40.
